### PR TITLE
[1.0] [DOC] SSLOptions, Utils::Headers, Utils::ParamsHash

### PR DESCRIPTION
**Pull request by olleolleolle**
## Description

This PR adds docs about API things.

## Todos
List any remaining work that needs to be done, i.e:
- [x] Have better wording
- [x] SSLOptions: Be exact with all datatypes

### CODE REVIEW

File: lib/faraday/options.rb [hunk 1]
```diff
@@ -218,14 +218,56 @@ def stream_response?
     end
   end
 
+  # SSL-related options.
+  #
+  # @!attribute verify
+  #   @return [Boolean] whether to very SSL certificates
```
Comment (iMacTia) [inline lib/faraday/options.rb]:
```suggestion
  #   @return [Boolean] whether to verify SSL certificates or not
```

File: lib/faraday/options.rb [hunk 2]
```diff
@@ -218,14 +218,56 @@ def stream_response?
...
+  #
+  # @!attribute ca_file
+  #   @return [String] CA file
+  #
+  # @!attribute ca_path
+  #   @return [String] CA path
+  #
+  # @!attribute verify_mode
+  #   @return [String] TODO: Explain this
```
Comment (iMacTia) [inline lib/faraday/options.rb]:
```suggestion
  #   @return [String] Any `OpenSSL::SSL::` constant (see https://ruby-doc.org/stdlib-2.5.1/libdoc/openssl/rdoc/OpenSSL/SSL.html)
```

File: lib/faraday/options.rb [hunk 3]
```diff
@@ -218,14 +218,56 @@ def stream_response?
...
+  #
+  # @!attribute cert_store
+  #   @return [Object] certificate store
```
Comment (iMacTia) [inline lib/faraday/options.rb]:
```suggestion
  #   @return [OpenSSL::X509::Store] certificate store
```

File: lib/faraday/options.rb [hunk 4]
```diff
@@ -218,14 +218,56 @@ def stream_response?
...
+  #
+  # @!attribute client_cert
+  #   @return [Object] client certificate
```
Comment (iMacTia) [inline lib/faraday/options.rb]:
```suggestion
  #   @return [String, OpenSSL::X509::Certificate] client certificate
```

File: lib/faraday/options.rb [hunk 5]
```diff
@@ -218,14 +218,56 @@ def stream_response?
...
+  #
+  # @!attribute client_key
+  #   @return [Object] client key
```
Comment (iMacTia) [inline lib/faraday/options.rb]:
```suggestion
  #   @return [String, OpenSSL::PKey::RSA, OpenSSL::PKey::DSA] client key
```

File: lib/faraday/options.rb [hunk 6]
```diff
@@ -218,14 +218,56 @@ def stream_response?
...
+  #
+  # @!attribute certificate
+  #   @return [Object] certificate
```
Comment (iMacTia) [inline lib/faraday/options.rb]:
```suggestion
  #   @return [OpenSSL::X509::Certificate] certificate (Excon only)
```

File: lib/faraday/options.rb [hunk 7]
```diff
@@ -218,14 +218,56 @@ def stream_response?
...
+  #
+  # @!attribute private_key
+  #   @return [Object] private key
```
Comment (iMacTia) [inline lib/faraday/options.rb]:
```suggestion
  #   @return [OpenSSL::PKey::RSA, OpenSSL::PKey::DSA] private key (Excon only)
```

File: lib/faraday/options.rb [hunk 8]
```diff
@@ -218,14 +218,56 @@ def stream_response?
...
+  #
+  # @!attribute verify_depth
+  #   @return [Object] verify depth
```
Comment (iMacTia) [inline lib/faraday/options.rb]:
```suggestion
  #   @return [Integer] maximum depth for the certificate chain verification
```

File: lib/faraday/options.rb [hunk 9]
```diff
@@ -218,14 +218,56 @@ def stream_response?
...
+  #
+  # @!attribute version
+  #   @return [Object] version
```
Comment (iMacTia) [inline lib/faraday/options.rb]:
```suggestion
  #   @return [String, Symbol] SSL version (see https://ruby-doc.org/stdlib-2.5.1/libdoc/openssl/rdoc/OpenSSL/SSL/SSLContext.html#method-i-ssl_version-3D)
```

File: lib/faraday/options.rb [hunk 10]
```diff
@@ -218,14 +218,56 @@ def stream_response?
...
+  #
+  # @!attribute min_version
+  #   @return [Object] minimum version
```
Comment (iMacTia) [inline lib/faraday/options.rb]:
```suggestion
  #   @return [String, Symbol] minimum SSL version (see https://ruby-doc.org/stdlib-2.5.1/libdoc/openssl/rdoc/OpenSSL/SSL/SSLContext.html#method-i-min_version-3D)
```

File: lib/faraday/options.rb [hunk 11]
```diff
@@ -218,14 +218,56 @@ def stream_response?
...
+  #
+  # @!attribute max_version
+  #   @return [Object] maximum version
```
Comment (iMacTia) [inline lib/faraday/options.rb]:
```suggestion
  #   @return [String, Symbol] maximum SSL version (see https://ruby-doc.org/stdlib-2.5.1/libdoc/openssl/rdoc/OpenSSL/SSL/SSLContext.html#method-i-max_version-3D)
```

**Comment by olleolleolle** [comment]
This is a super way of improving docs together.

**Pull request merged**